Description: CameraBin: support FlashVideoLight flash mode
 Apps in Ubuntu Touch tends to hardcode FlashVideoLight mode, as it's
 the only mode available with qtubuntu-camera backend. Add support for
 this mode for compatibility. Notes that if an app reads back the mode,
 we'll report back FlashTorch. However, AFAIK most apps don't read back
 the mode.
Author: Ratchanan Srirattanamet <ratchanan@ubports.com>
Forwarded: not-needed
Last-Update: 2020-06-02
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/plugins/gstreamer/camerabin/camerabinflash.cpp
+++ b/src/plugins/gstreamer/camerabin/camerabinflash.cpp
@@ -100,7 +100,9 @@
     gboolean torchEnabled = false;
     g_object_get(G_OBJECT(m_session->cameraSource()), "video-torch", &torchEnabled, NULL);
 
-    gboolean enableTorch = mode.testFlag(QCameraExposure::FlashTorch);
+    gboolean enableTorch = mode.testFlag(QCameraExposure::FlashTorch) ||
+                           // For compatibility with majority of UT apps.
+                           mode.testFlag(QCameraExposure::FlashVideoLight);
 
     if (bool(enableTorch) != bool(torchEnabled)) {
         g_object_set(G_OBJECT(m_session->cameraSource()), "video-torch", enableTorch, NULL);
